internal/transport/http2_server.go

```diff
@@ -1042,6 +1042,18 @@ func (t *http2Server) writeHeaderLocked(s *Stream) error {
 	return nil
 }
 
+type NoopFunc func()
+
+var noop NoopFunc = func() {
```

Comment:
please see how other noop functions are implemented. There are some for goaway tests. Please make a habit of going through existing code to get an idea.

Give meaningful name to it. May be something like SignalDeadlineExceededForTesting. Also, it should not be exported.